### PR TITLE
add zipf function to numpy frontend 

### LINKED ISSUE #22058: zipf

### PR DESCRIPTION
**Pull request by Mwangi-ke**
Close #22058

### CODE REVIEW

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

File: ivy/functional/frontends/numpy/random/functions.py [hunk 1]
```diff
@@ -233,3 +233,14 @@ def triangular(left, mode, right, size=None):
...
+
+@to_ivy_arrays_and_back
+@from_zero_dim_arrays_to_scalar
+def zipf(a, size=None):
+    if a <= 1:
+        raise ValueError(
+            "Parameter 'alpha' must be greater than 1 for the Zipf distribution."
+        )
+    return ivy.floor((1.0 / a) ** 1.0).astype(int)
+
```
Comment (hmahmood24) [inline ivy/functional/frontends/numpy/random/functions.py:246]: Can you please state what's your reference for this implementation? Maybe a link to any doc/paper etc? I'm not sure if this is how the `zipf` distribution is calculated.

**Review by hmahmood24 — CHANGES_REQUESTED** [review]
Hey @Mwangi-ke, thanks a lot for the PR! However, looking more closely, it’s unclear how this PR would meaningfully contribute towards the project, because of the following points:

1. You haven't added a test for your implementation and without that we cannot proceed with this PR so please ensure that you implement a test for this function (the same way the tests are added for all frontend functions in `ivy`) and ensure that the test passes for all backends.
2. There are some [lint/formatting issues](https://github.com/unifyai/ivy/actions/runs/5888185815/job/15968890813?pr=22063) with this PR as well. Can you kindly fix those as well?

Please feel free to take a look at our [Contributing](https://unify.ai/docs/ivy/overview/contributing.html) and [Deep Dive](https://unify.ai/docs/ivy/overview/deep_dive.html) guides first. Please ask for a re-review once you've dealt with the points above. Thanks! :)

File: ivy/functional/frontends/numpy/random/functions.py (hunk 1 above)
Comment (Mwangi-ke) [reply ivy/functional/frontends/numpy/random/functions.py:246]: Hey @hmahmood24 ,thanks for your review.                                                                                           This is the link to the documentation  on zipf function https://numpy.org/doc/stable/reference/random/generated/numpy.random.zipf.html                                                                                                          This is the Test for the zipf function  https://github.com/unifyai/ivy/pull/22061 which is also another PR. Advise on how I may merge them also if possible.

Comment (Mwangi-ke) [reply ivy/functional/frontends/numpy/random/functions.py:246]: i have had to push my commits under a new branch so as to fix all issues as advised. I however had to create a new pull request with the following link   https://github.com/unifyai/ivy/pull/22424
It contains both the zipf function and the test. Should i close this PR . Advise please

**Pull request closed without merging**